executor/show_placement.go

```diff
@@ -421,39 +442,55 @@ func (e *ShowExec) getPolicyPlacement(policyRef *model.PolicyRefInfo) (settings
 	return policy.PlacementSettings, nil
 }
 
-func fetchScheduled(ctx context.Context, scheduleState map[int64]bool, id int64) (bool, error) {
+func fetchScheduleState(ctx context.Context, scheduleState map[int64]placementScheduleState, id int64) (placementScheduleState, error) {
 	if s, ok := scheduleState[id]; ok {
 		return s, nil
 	}
 	startKey := codec.EncodeBytes(nil, tablecodec.GenTablePrefix(id))
 	endKey := codec.EncodeBytes(nil, tablecodec.GenTablePrefix(id+1))
 	schedule, err := infosync.GetReplicationState(ctx, startKey, endKey)
+	var state placementScheduleState
+	switch schedule {
+	case "\"REPLICATED\"\n":
```

Comment:
It's strange...

Reply:
This is the only entry, it is overkill to parse the json. It is certain that there are only three possibilities.

Reply:
But yeah, maybe deserialize it in `GetReplicationState`.

Reply:
> This is the only entry, it is overkill to parse the json. It is certain that there are only three possibilities.

I don't think so... Without JSON parser, we must make an assumption that the return value equal to `"REPLICATED"\n` . But `"REPLICATED"\n\n` and `"REPLICATED"` are still legal values and the API maintainer may not be aware it will cause a break when some day it changes to the above format. 

In other words, it is safer to use a JSON parser than string compare

Reply:
I believe the API maintainer is me. And pretty much every JSON marshal of golang, will results into `"..."` for a string. Then PD will add another '\n' automatically, which has been around for years. If you really insist so, `StartsWith` is enough to ignore the unknown suffix.